### ISSUE TITLE
Fix `character-reference` reports error about correct code

### ISSUE THREAD

**Issue by kagankan** [markuplint/markuplint#563]
- Markuplint version: `2.11.0`
- Parser lang: HTML 
- Node.js version: `16.16.0`
- OS: Windows

## Describe the bug

`character-reference` rule reports error about some correct code.

## Code Example

```html
<img src="" alt="<>" />
<p>"text"</p>
```

## Steps To Reproduce

Lint HTML file including the `Code Example` with  `character-reference` rule

## The current behavior

`&`, `<`, `>`, `"` are always reported as invalid character

## The expected behavior

As my understanding, escaping is only required under the following conditions.

`<`: outside attribute values
`>`: in non-quoted attribute values
`"`: in attribute values quoted by `"`
`'`:  in attribute values quoted by `'`
`&`: always

NOTE: Here is my article about escaping characters in HTML (Japanese)
https://zenn.dev/kagan/articles/4038e7a9a5ff5e

So, `<img src="" alt="<>" />` and `<p>"text"</p>` are correct codes, I think.

(Sorry if I misunderstood.)

……However, I can understand that it is safe to escape all these characters.

**Comment by YusukeHirao** [comment]
@kagankan 

Thank you for your report.

You're right. This rule doesn't evaluate the characters strictly. It only evaluates efficiently.
This current rule is easy for everybody to understand. But the strict evaluation is complex for the rule algorithm and humans.  And it's so difficult for me to improve the program.

I will not change this behavior. But I'm going to add the description "it is not strict evaluation" to the document. And I will move to code style as a category because it may not be conformance checking.

Thank you.

**Comment by YusukeHirao** [comment]
Tasks:

- [ ] Add to the document.
- [ ] Move category.
